modules/percolator/src/main/java/org/elasticsearch/percolator/ExtractQueryTermsService.java

```diff
@@ -376,6 +378,16 @@ static Result extractQueryTerms(Query query) {
         };
     }
 
+    static Function<Query, Result> functionScoreQuery() {
+        return query -> {
+            Result result = extractQueryTerms(((FunctionScoreQuery) query).getSubQuery());
+            // If min_score is specified we can't guarantee upfront that this percolator query always matches
+            // if it matches with the percolator document matches with the extracted terms.
+            // Min score filters out docs, which is different than the functions, which just influences the score.
+            return new Result(false, result.terms);
```

Comment:
can't we add a getter for FunctionScoreQuery.minScore and use it to know whether it is confirmed or not?